example/open-new-position-lp.ts

```diff
@@ -0,0 +1,111 @@
+
+  // `minLPAmount` is for slippage control
+  // by specifying it as `0` this would mean the user is willing to accept any amount of LP token received
+  // this could proven to be dangerous for price manipulation attack, we put it as `0` here for the simplicity of the code example
+  // Alpaca Finance frontend would calculate this value based on the current price with some percentage of slippage tolerance
+  const minLPAmount = '0'
+
+  // In this example, we are opening a new position on ALPACA-BUSD PancakeSwap Farm with BUSD as borrowing asset
+  // We will farm this position with our 100 BUSD and 100 ALPACA principal and we will borrow 100 BUSD from Alpaca Finance's Lending Vault
+  const inputBaseTokenAmount = ethers.utils.parseEther('100')
+  const inputFarmingTokenAmount = ethers.utils.parseEther('100')
+  console.log('inputFarmingTokenAmount', inputFarmingTokenAmount)
+  const borrowAmount = ethers.utils.parseEther('100')
+
+  //  _____                               
+  // |  __ \                              
+  // | |__) | __ ___ _ __   __ _ _ __ ___ 
+  // |  ___/ '__/ _ \ '_ \ / _` | '__/ _ \
+  // | |   | | |  __/ |_) | (_| | | |  __/
+  // |_|   |_|  \___| .__/ \__,_|_|  \___|
+  //                | |                   
+  //                |_|                   
+  const vault = Vault__factory.connect(vaultAddress, signer)
+  const worker = IWorker__factory.connect(workerAddress, signer)
+  const baseToken = BEP20__factory.connect(await worker.baseToken(), signer)
+  const farmingToken = BEP20__factory.connect(
+    await worker.farmingToken(),
+    signer,
+  )
+
+  const baseTokenAllowance = await baseToken.allowance(
+    signer.address,
+    vaultAddress,
+  )
+  if (baseTokenAllowance.isZero()) {
+    await baseToken.approve(vaultAddress, constants.MaxUint256)
+  }
+  const farmingTokenAllowance = await farmingToken.allowance(
+    signer.address,
+    vaultAddress,
+  )
+  if (!inputFarmingTokenAmount.isZero() && farmingTokenAllowance.isZero()) {
+    await farmingToken.approve(vaultAddress, constants.MaxUint256)
+  }
+    
+  // __          __        _    
+  // \ \        / /       | |   
+  //  \ \  /\  / /__  _ __| | __
+  //   \ \/  \/ / _ \| '__| |/ /
+  //    \  /\  / (_) | |  |   < 
+  //     \/  \/ \___/|_|  |_|\_\
+
+  const nextPositionID = await vault.nextPositionID()
+
+  const encodedStrategyParams = new TwoSideOptimalStrategy().encodeStrategyParams(inputFarmingTokenAmount.toString(), minLPAmount)
+    console.log('encodedStrategyParams', encodedStrategyParams)
+  const result = await vault.work(
+    0,
+    workerAddress,
+    inputBaseTokenAmount,
+    borrowAmount,
+    '0',
+    ethers.utils.defaultAbiCoder.encode(
+      ['address', 'bytes'],
+      [strategyAddress, encodedStrategyParams],
+    ),
+    {
+      gasLimit: OpenPositionGasLimit,
+      value: '0',
+    },
+  )
```

Comment:
Sorry you are using js / ts library to interact with Alpaca contract here. While our requirement is to use contract to contract interaction.

Will the following code work properly to open a new position?
```javascript
alpacaVault.work(
    0,
    alpacaWorkerAddress,
    principalAmount,
    borrowAmount,
    0,
    abi.encode(
        alpacaStrategyAddress,
        abi.encode(
            farmingAmount,
            minimumLPAmount
        )
    )
);
```

Reply:
Something along that line should work. I'll try doing a test with contract to contract call.